src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -45,11 +48,26 @@ internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F,
             var param = F.Parameter(Parameters[0]);
             foreach (var field in ContainingType.GetFieldsToEmit())
             {
-                if (!field.IsStatic)
+                if (!field.IsStatic && field.ContainingType.Equals(ContainingType))
                 {
                     statements.Add(F.Assignment(F.Field(F.This(), field), F.Field(param, field)));
                 }
             }
         }
+
+        internal static MethodSymbol? FindCopyConstructor(NamedTypeSymbol containingType)
+        {
+            foreach (var member in containingType.InstanceConstructors)
+            {
+                if (member is MethodSymbol { IsStatic: false, ParameterCount: 1, Arity: 0 } method &&
+                    method.Parameters[0].Type.Equals(containingType, TypeCompareKind.CLRSignatureCompareOptions) &&
+                    method.Parameters[0].RefKind == RefKind.None)
+                {
+                    return method;
```

Comment:
>return method; [](start = 20, length = 14)

I think we should check accessibility before declaring the method a valid candidate.

Reply:
I left the accessibility check to be the responsibility of the caller of `FindCopyConstructor`. That is more consistent with existing code (`GenerateBaseParameterlessConstructorInitializer` for example).

---
In reply to: [438764948](https://github.com/dotnet/roslyn/pull/45022#discussion_r438764948) [](ancestors = 438764948)

Reply:
> I left the accessibility check to be the responsibility of the caller of FindCopyConstructor. That is more consistent with existing code (GenerateBaseParameterlessConstructorInitializer for example).


This is wrong. Accesibility check belongs here and should come first

---
In reply to: [438934812](https://github.com/dotnet/roslyn/pull/45022#discussion_r438934812) [](ancestors = 438934812,438764948)

Reply:
Binder is not needed for accessibility check, see MethodCompiler.GenerateBaseParameterlessConstructorInitializer

---
In reply to: [438941054](https://github.com/dotnet/roslyn/pull/45022#discussion_r438941054) [](ancestors = 438941054,438934812,438764948)